### PR TITLE
Allow empty map settings

### LINKED ISSUE #9247: Empty settings cause map settings error
Currently empty settings cause map settings errors. For example, if you open [`ddnet-maps/types/solo/maps/Masquerade.map`](https://github.com/ddnet/ddnet-maps/blob/ba0d6d44d965812a708051bfa6c54ae8da606970/types/solo/maps/Masquerade.map) in the editor, you receive the following screen:

screenshot_2024-11-14_11-58-38

I don't think this is a map-issue, because this is used here to put structure into the tuning regions and should be ignored by the editor as @Robyt3 suggested.

(Also the map still contains bullshit settings, but this is another issue)

### PR DESCRIPTION
**Pull request by KebsCS**
Closes #9247

## Checklist

- [x] Tested the change ingame
- [ ] Provided screenshots if it is a visual change
- [ ] Tested in combination with possibly related configuration options
- [ ] Written a unit test (especially base/) or added coverage to integration test
- [ ] Considered possible null pointers and out of bounds array indexing
- [ ] Changed no physics that affect existing maps
- [ ] Tested the change with [ASan+UBSan or valgrind's memcheck](https://github.com/ddnet/ddnet/#using-addresssanitizer--undefinedbehavioursanitizer-or-valgrinds-memcheck) (optional)

### CODE REVIEW

File: src/game/editor/editor_server_settings.cpp [hunk 1]
```diff
@@ -1675,7 +1675,7 @@ void CMapSettingsBackend::CContext::UpdatePossibleMatches()
 	m_vPossibleMatches.clear();
 	m_DropdownContext.m_Selected = -1;
 
-	if(m_CommentOffset == 0)
+	if(m_CommentOffset == 0 || m_aCommand[0] == '\0')
```
Comment (heinrich5991) [inline src/game/editor/editor_server_settings.cpp:1678]: Does this do something other allowing empty map settings? If so, please add it to the commit description.

Comment (KebsCS) [reply src/game/editor/editor_server_settings.cpp:1678]: It just allows empty map settings. `m_aCommand` holds the setting and simply returns if it's empty, similar to an entire line comment

File: src/game/editor/editor_server_settings.cpp [hunk 2]
```diff
@@ -1870,14 +1870,14 @@ int CMapSettingsBackend::CContext::CheckCollision(const char *pInputString, cons
...
 	{ // Ignore comments
 		Result = ECollisionCheckResult::ADD;
 		return -1;
 	}
 
-	const int InputLength = str_length(pInputString);
-
 	struct SArgument
 	{
 		char m_aValue[128];
```
Comment (AssassinTee) [inline src/game/editor/editor_server_settings.cpp:1883]: below here `v` is yet another check for `if(InputLength == 0)`, I guess it can be removed

**Review by Robyt3 — APPROVED** [review]

**Review by AssassinTee — APPROVED** [review]

File: src/game/editor/editor_server_settings.cpp (hunk 1 above)
Comment (heinrich5991) [reply src/game/editor/editor_server_settings.cpp:1678]: From the code I'd say it omits suggestions when you are at the start of an empty line. Is that so?

Comment (KebsCS) [reply src/game/editor/editor_server_settings.cpp:1678]: There weren’t any suggestions with an empty line before this anyway

Comment (heinrich5991) [reply src/game/editor/editor_server_settings.cpp:1678]: What does the change on this line do, then?

Comment (KebsCS) [reply src/game/editor/editor_server_settings.cpp:1678]: The same as three comments above, now that empty settings are valid, this function would incorrectly try to update matches for them. As mentioned before, there are no matches for an empty string

Comment (Robyt3) [reply src/game/editor/editor_server_settings.cpp:1678]: > As mentioned before, there are no matches for an empty string

But it was previously possible to press Ctrl+Space to show the suggestions even when the input is empty. That's not possible anymore with these changes.

**Pull request merged**